include/bluetooth/services/ras.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef BT_RAS_H_
+#define BT_RAS_H_
+
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/uuid.h>
+
+/** @file
+ *  @defgroup bt_ras Ranging Service API
+ *  @{
+ *  @brief API for the Ranging Service (RAS).
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief UUID of the Ranging Service. **/
+#define BT_UUID_RANGING_SERVICE_VAL (0x185B)
+
+/** @brief UUID of the RAS Features Characteristic. **/
+#define BT_UUID_RAS_FEATURES_VAL (0x2C14)
+
+/** @brief UUID of the Real-time Ranging Data Characteristic. **/
+#define BT_UUID_RAS_REALTIME_RD_VAL (0x2C15)
+
+/** @brief UUID of the On-demand Ranging Data Characteristic. **/
+#define BT_UUID_RAS_ONDEMAND_RD_VAL (0x2C16)
+
+/** @brief UUID of the RAS Control Point Characteristic. **/
+#define BT_UUID_RAS_CP_VAL (0x2C17)
+
+/** @brief UUID of the Ranging Data Ready Characteristic. **/
+#define BT_UUID_RAS_RD_READY_VAL (0x2C18)
+
+/** @brief UUID of the Ranging Data Overwritten Characteristic. **/
+#define BT_UUID_RAS_RD_OVERWRITTEN_VAL (0x2C19)
+
+#define BT_UUID_RANGING_SERVICE    BT_UUID_DECLARE_16(BT_UUID_RANGING_SERVICE_VAL)
+#define BT_UUID_RAS_FEATURES       BT_UUID_DECLARE_16(BT_UUID_RAS_FEATURES_VAL)
+#define BT_UUID_RAS_REALTIME_RD    BT_UUID_DECLARE_16(BT_UUID_RAS_REALTIME_RD_VAL)
+#define BT_UUID_RAS_ONDEMAND_RD    BT_UUID_DECLARE_16(BT_UUID_RAS_ONDEMAND_RD_VAL)
+#define BT_UUID_RAS_CP             BT_UUID_DECLARE_16(BT_UUID_RAS_CP_VAL)
+#define BT_UUID_RAS_RD_READY       BT_UUID_DECLARE_16(BT_UUID_RAS_RD_READY_VAL)
+#define BT_UUID_RAS_RD_OVERWRITTEN BT_UUID_DECLARE_16(BT_UUID_RAS_RD_OVERWRITTEN_VAL)
+
+#define BT_RAS_MAX_SUBEVENTS_PER_PROCEDURE 32
```

Comment:
Yes, it's used for buffer sizing and it does end up at around 10kB (9476 bytes):
```
#define BT_RAS_PROCEDURE_MEM (BT_RAS_RANGING_HEADER_LEN \
			     + (BT_RAS_MAX_SUBEVENTS_PER_PROCEDURE * BT_RAS_SUBEVENT_HEADER_LEN) \
			     + (BT_RAS_MAX_STEPS_PER_PROCEDURE * BT_RAS_STEP_MODE_LEN) \
			     + (BT_RAS_MAX_STEPS_PER_PROCEDURE * BT_RAS_MAX_STEP_DATA_LEN))
```
The spec says 
> The RAS Server shall have memory to record at least one complete Ranging Data for one CS Procedure.

But as long as the defaults satisfy this, I think making it configurable makes sense.

Reply:
I think that sentence is still satisfied even without the maximum possible procedure size, don't you think so ? "one complete Ranging Data for one CS procedure" can also be a CS procedure with a single subevent and a small number of steps (at least the way I read it), as this would be complete ranging data

Reply:
It's a bit vague as to what "complete" means, I read it as any valid ranging data procedure the peer might send, including one with the maximum number of steps.
I still agree it makes sense for this to be configurable, since the app will have better expectations on the data length it expects.
Do you think the default should not be 32/256?